cubelib/src/cube/cube_corners.rs

```diff
@@ -229,6 +232,62 @@ impl Default for CubeCornersOdd {
     }
 }
 
+impl CubeCornersOdd {
+    #[inline]
+    #[cfg(target_feature = "avx2")]
+    pub fn random(parity: bool, rng: &mut ThreadRng) -> Self {
+        let bytes = random_corners(parity, rng);
+        unsafe { avx2::unsafe_from_bytes(bytes) }
+    }
+
+    #[inline]
+    #[cfg(all(target_arch = "wasm32", not(target_feature = "avx2")))]
+    pub fn random(parity: bool, rng: &mut ThreadRng) -> Self {
+        let bytes = random_corners(parity, rng);
+        wasm32a::from_bytes(bytes)
+    }
+
+    #[inline]
+    #[cfg(all(target_feature = "neon", not(target_feature = "avx2")))]
+    pub fn random(parity: bool, rng: &mut ThreadRng) -> Self {
+        let bytes = random_corners(parity, rng);
+        unsafe { neon::unsafe_from_bytes(bytes) }
+    }
+}
+
+fn random_corners(parity: bool, rng: &mut ThreadRng) -> [u8; 8] {
+    let mut corner_bytes: [u8; 8] = [0, 1, 2, 3, 4, 5, 6, 7];
+    let mut orientation_parity = 0;
+    let mut swap_parity = false;
+
+    fn get_bytes(piece_id: u8, orientation: u8) -> u8 {
+        (piece_id << 5) | (orientation & 0x3)
+    }
+
+    for i in 0..6 {
+        let swap_index = rng.gen_range(i..8);
```

Comment:
Swapping with `i..N` an unbiased shuffle. Doing it manually to keep track of parity